src/megaapi_impl.cpp

```diff
@@ -13085,43 +13063,51 @@ void MegaApiImpl::folderlinkinfo_result(error e, handle owner, handle /*ph*/, st
 }
 
 #ifdef ENABLE_SYNC
-void MegaApiImpl::syncupdate_state(int tag, syncstate_t newstate, SyncError syncerror, bool fireDisableEvent)
+
+MegaSyncPrivate* MegaApiImpl::cachedMegaSyncPrivateByTag(int tag)
 {
-    auto syncpair = syncMap.find(tag);
-    if( syncpair == syncMap.end())
+    if (tag == mCachedMegaSyncPrivateTag && mCachedMegaSyncPrivate)
```

Comment:
Why not:
```suggestion cpp
    if (mCachedMegaSyncPrivate && tag == mCachedMegaSyncPrivate.getTag())
```
That way, we can directly get rid of the `mCachedMegaSyncPrivateTag`.
I'll fix it ;)